### PR TITLE
TDL-17904: Implement request timeout

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-17904: Implement request timeout
- Added timeout for requests with a default value of 300 seconds.
- Added backoff for `Timeout` error

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_tiktok_ads/__init__.py [hunk 1]
```diff
@@ -21,7 +21,8 @@ def main():
     args = utils.parse_args(REQUIRED_CONFIG_KEYS)
 
     with TikTokClient(access_token=args.config['access_token'],
-                      user_agent=args.config['user_agent']) as tik_tok_client:
+                      user_agent=args.config['user_agent'],
+                      request_timeout=args.config.get('request_timeout')) as tik_tok_client:
```
Comment (savan-chovatiya) [inline tap_tiktok_ads/__init__.py:25]: Add this new config variable in the README file and sample config file in the repository.

Comment (harshpatel4crest) [reply tap_tiktok_ads/__init__.py:25]: Updated README file.

File: tests/unittests/test_timeout.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        # set config
+        config = {
+            "access_token": "test_access_token",
+            "user_agent": "test_user_agent"
+        }
+
+        # verify that we raise Timeout error when using "with" statement
+        with self.assertRaises(requests.Timeout):
+            # create client and call function
+            client = TikTokClient(config.get("access_token"), config.get("user_agent"), config.get("request_timeout"))
```
Comment (umangagrawal-crest) [inline tests/unittests/test_timeout.py:290]: Move this client initialization out of the `with` statement, so it makes sure that the error is raised from the client.request.
Also, make same changes in other functions wherever it applies

Comment (harshpatel4crest) [reply tests/unittests/test_timeout.py]: Updated.

**Review by savan-chovatiya — APPROVED** [review]

**Review by umangagrawal-crest — APPROVED** [review]

File: tests/unittests/test_timeout.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+# mocked response class
+class Mockresponse:
+    def __init__(self, status_code, json, raise_error, headers=None):
+        self.status_code = status_code
+        self.raise_error = raise_error
+        self.text = json
+        self.headers = headers
+
+    def raise_for_status(self):
```
Comment (dbshah1212) [inline tests/unittests/test_timeout.py:14]: Do add function comments.

Comment (harshpatel4crest) [reply tests/unittests/test_timeout.py:14]: Added function comments.

**Review by dbshah1212 — APPROVED** [review]

**Pull request merged**